packages/integrations/netlify/src/index.ts

```diff
@@ -697,12 +702,25 @@ export default function netlifyIntegration(
 				if (existingSessionModule) {
 					server.moduleGraph.invalidateModule(existingSessionModule);
 				}
-				server.middlewares.use((req, _res, next) => {
+				server.middlewares.use((req, res, next) => {
 					const locals = Symbol.for('astro.locals');
+					const promises: Array<Promise<unknown>> = [];
 					Reflect.set(req, locals, {
 						...Reflect.get(req, locals),
-						netlify: { context: getLocalDevNetlifyContext(req) },
+						netlify: { context: getLocalDevNetlifyContext(req, promises) },
+					});
+
+					// Wait for response to finish, then resolve waitUntil promises
+					res.on('finish', async () => {
+						if (promises.length > 0) {
+							try {
+								await Promise.allSettled(promises);
```

Comment:
Does this actually do anything? Does anything actually await this handler? I wonder if `waitUntil` should just call `queueMicrotask`